VideoWeb/VideoWeb/ClientApp/src/app/shared/mappers/unread-messages-model-mapper.spec.ts

```diff
@@ -0,0 +1,42 @@
+import { UnreadAdminMessageResponse } from 'src/app/services/clients/api-client';
+import { UnreadAdminMessageModel } from '../../waiting-space/models/unread-admin-message-model';
+import { ConferenceTestData } from 'src/app/testing/mocks/data/conference-test-data';
+import { UnreadAdminMessageModelMapper } from './unread-messages-model-mapper';
+
+describe('ParticipantPanelModelMapper', () => {
+    let mapper: UnreadAdminMessageModelMapper;
+    const conference = new ConferenceTestData().getConferenceDetailNow();
+    const unreadMessagesCount = 5;
+
+    beforeEach(() => {
+        mapper = new UnreadAdminMessageModelMapper();
+    });
+
+    it('should map unread messages response array to conference', () => {
+        // arrange
+        const unreadMessages = conference.participants.map(
```

Comment:
Isn't this already doing what the mapper does? Shouldn't you be passing in an array of UnreadAdminMessageResponse and not the model?

Reply:
The mapper is outside the scope of these unit tests, this is just for a return value for the spy.

Reply:
well spotted there, I am not sure how even the mapper didnt explode

Reply:
The ~~pro~~ of having strict typing turned off ;)

Reply:
The model derives from the base type so is actually castable to it. Even with strict typing this wouldn't of been picked up ;)